packit.spec

```diff
@@ -70,20 +39,21 @@ check out packit package for the executable.
 
 %prep
 %autosetup -n %{pypi_name}-%{version}
-# Remove bundled egg-info
-rm -rf %{pypi_name}.egg-info
-
-%if 0%{?rhel}
-# rhbz#1968618 still not fixed for epel-8
-sed -i -e 's|koji|# koji|' setup.cfg
+%generate_buildrequires
```

Comment:
```suggestion


%generate_buildrequires
```

Reply:
Why the excessive whitespace?

Reply:
It's not excessive, separating sections with two empty lines is recommended and it helps readability (just like two blank lines between top-level function and class definitions dictated by PEP8).

Reply:
Ok, but this is still in the `%prep` section, not a separate one like `%build`. That's what I was finding confusing

Reply:
It's not, `%generate_buildrequires` is a section, not a macro.

Reply:
I'm sorry but I find this confusing. When I do `rpmspec --parse packit.spec` I get:

<details><summary> rpmspec --parse packit.spec</summary>

```
%prep
## autosetup -n packitos-0.73.0.post10+ga7c018a.d20230410

cd '/home/lecris/rpmbuild/BUILD'
rm -rf 'packitos-0.73.0.post10+ga7c018a.d20230410'
/usr/lib/rpm/rpmuncompress -x /home/lecris/rpmbuild/SOURCES/packitos-0.73.0.post10+ga7c018a.d20230410.tar.gz
STATUS=$?
if [ $STATUS -ne 0 ]; then
  exit $STATUS
fi
cd 'packitos-0.73.0.post10+ga7c018a.d20230410'
/usr/bin/chmod -Rf a+rX,u+w,g-w,o-w .


## pyproject_buildrequires -x testing
\

# The _auto_set_build_flags feature does not do this in %generate_buildrequires section,
# but we want to get an environment consistent with %build:

  CFLAGS="${CFLAGS:--O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASS
ERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -m64  -mtune=generic -fasynchronous-unwind
-tables -fstack-clash-protection -fcf-protection}" ; export CFLAGS ; 
  CXXFLAGS="${CXXFLAGS:--O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX
_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -m64  -mtune=generic -fasynchronous-un
wind-tables -fstack-clash-protection -fcf-protection}" ; export CXXFLAGS ; 
  FFLAGS="${FFLAGS:--O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASS
ERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -m64  -mtune=generic -fasynchronous-unwind
-tables -fstack-clash-protection -fcf-protection -I/usr/lib64/gfortran/modules}" ; export FFLAGS ; 
  FCFLAGS="${FCFLAGS:--O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_A
SSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -m64  -mtune=generic -fasynchronous-unwi
nd-tables -fstack-clash-protection -fcf-protection -I/usr/lib64/gfortran/modules}" ; export FCFLAGS ; 
  LDFLAGS="${LDFLAGS:--Wl,-z,relro -Wl,--as-needed  -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -Wl,--build-i
d=sha1 }" ; export LDFLAGS ; 
  LT_SYS_LIBRARY_PATH="${LT_SYS_LIBRARY_PATH:-/usr/lib64:}" ; export LT_SYS_LIBRARY_PATH ; 
  CC="${CC:-gcc}" ; export CC ; 
  CXX="${CXX:-g++}" ; export CXX
# The default flags expect the package note file to exist
# see https://bugzilla.redhat.com/show_bug.cgi?id=2097535


echo 'pyproject-rpm-macros'  # first stdout line matches the implementation in macros.aaa-pyproject-srpm
echo 'python3-devel'
echo 'python3dist(pip) >= 19'
echo 'python3dist(packaging)'
if [ -f pyproject.toml ]; then
  echo '(python3dist(toml) if python3-devel < 3.11)'
elif [ -f setup.py ]; then
  # Note: If the default requirements change, also change them in the script!
  echo 'python3dist(setuptools) >= 40.8'
  echo 'python3dist(wheel)'
else
  echo 'ERROR: Neither pyproject.toml nor setup.py found, consider using %pyproject_buildrequires -N <requirements-file> if this is not a Python package.' >&2
  exit 1
fi
# setuptools assumes no pre-existing dist-info
rm -rfv *.dist-info/ >&2
if [ -f /usr/bin/python3 ]; then
  mkdir -p "/home/lecris/rpmbuild/BUILD/packitos-0.73.0.post10+ga7c018a.d20230410/.pyproject-builddir"
  CFLAGS="${CFLAGS:-${RPM_OPT_FLAGS}}" LDFLAGS="${LDFLAGS:-${RPM_LD_FLAGS}}" TMPDIR="/home/lecris/rpmbuild/BUILD/packitos-0.73.0.post10+ga7c018a.d20230410/.pyproject-builddir" \
  RPM_TOXENV="py311" HOSTNAME="rpmbuild" /usr/bin/python3 -Bs /usr/lib/rpm/redhat/pyproject_buildrequires.py --generate-extras --python3_pkgversion 3 --wheeldir /home/lecris/rpmbuild/BUILD/packitos-0.73.0.post10+ga7c018a.d20230410/pyproject-wheeldir -x testing
fi
# Incomplete .dist-info dir might confuse importlib.metadata
rm -rfv *.dist-info/ >&2


%build
```

</details>

I have added comments to the corresponding parts that were expanded. Indeed I do not see `%generate_buildrequires` to be expanded like a macro, but also I don't see it to be its own section like `%prep` or `%build`. Is `rpmspec` not able to detect it?

Reply:
Hm, I don't know if that's a bug in `rpmspec` that it doesn't seem to support it, but it's documented here:
https://rpm-software-management.github.io/rpm/manual/spec.html#generate_buildrequires-since-rpm--415

Reply:
Right, `%generate_buildrequires` is a separate section like `%prep` or `%build`. There should be at least one new line between sections.